core/src/main/scala/kafka/coordinator/GroupCoordinator.scala

```diff
@@ -454,23 +454,45 @@ class GroupCoordinator(val brokerId: Int,
     delayedOffsetStore.foreach(groupManager.store)
   }
 
+  def getPartitions(groupId: String): List[TopicPartition] = {
+      val group = groupManager.getGroup(groupId)
+      group.fold(List[TopicPartition]())(_.allOffsets.keySet.toList)
+  }
 
   def handleFetchOffsets(groupId: String,
-                         partitions: Seq[TopicPartition]): Map[TopicPartition, OffsetFetchResponse.PartitionData] = {
+                         partitions: Seq[TopicPartition],
+                         apiVersion: Short): (Map[TopicPartition, OffsetFetchResponse.PartitionData], Short) = {
+    require(apiVersion >= 1)
+
     if (!isActive.get) {
-      partitions.map { topicPartition =>
-        (topicPartition, new OffsetFetchResponse.PartitionData(OffsetFetchResponse.INVALID_OFFSET, "", Errors.GROUP_COORDINATOR_NOT_AVAILABLE.code))}.toMap
+      (apiVersion match {
```

Comment:
I think I'm missing something here. We are already iterating over all partitions here (for version 1) and injecting the proper error code. If we want to do the injection in `OffsetFetchResponse` constructor, we need to iteration over them again, which wouldn't be very efficient. Could you please clarify? Thanks.